File: PHP/Memory.php
<?php

class Memory extends Hardware {

    /* Private Attribute */
    private $frequency = "Frequency";
    private $memorySize = "Memory Size";
    private $supportsCuda = "Supports Cuda";

    /* Constructor */
    public function __construct($frequency = "Frequency", $memorySize = "Memory Size", $supportsCuda = "Supports Cuda") {
        $this->frequency = $frequency;
        $this->memorySize = $memorySize;
        $this->supportsCuda = $supportsCuda;
    }

    /* Setter dan Getter Frequency */
    public function setFrequency($frequency) {
        $this->frequency = $frequency;
    }

    public function getFrequency() {
        return $this->frequency;
    }

    /* Setter dan Getter Memory Size */
    public function setMemorySize($memorySize) {
        $this->memorySize = $memorySize;
    }

    public function getMemorySize() {
        return $this->memorySize;
    }

    /* Setter dan Getter Supports Cuda */
    public function setSupportsCuda($supportsCuda) {
        $this->supportsCuda = $supportsCuda;
    }

    public function getSupportsCuda() {
        return $this->supportsCuda;
    }

    /* Menampilkan data Memory */
    public function showMemoryData() {
        echo "<br /> __/Product Data___________________";
        echo "<br /> Price : ". $this->getPrice();
        echo "<br /> Id Product : ". $this->getIdProduct();
        echo "<br /> __________________________________";
        echo "<br />";
        echo "<br /> __/Hardware Data___________________";
        echo "<br /> Brand : ". $this->getBrand();
        echo "<br /> Model : ". $this->getModel();
        echo "<br /> __________________________________";
        echo "<br />";
        echo "<br /> __/Memory Data___________________";
        echo "<br /> Frequency : ". $this->getFrequency();
        echo "<br /> Memory Size : ". $this->getMemorySize();
        echo "<br /> Supports Cuda : ". $this->getSupportsCuda();
        echo "<br /> __________________________________";
        echo "<br />";
    }

    /* Destructor */
    public function __destruct() {

	}

}

?>

File: Java/Memory.java
public class Memory extends Hardware {

    /* Private Attribute */
    private String frequency;
    private String memorySize;
    private String supportsCuda;

    /* Constuctor */
    public Memory(String frequency, String memorySize, String supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public Memory() {

    }

    /* Setter dan Getter Frequency */
    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    public String getFrequency() {
        return frequency;
    }

    /* Setter dan Getter Memory Size */
    public void setMemorySize(String memorySize) {
        this.memorySize = memorySize;
    }

    public String getMemorySize() {
        return memorySize;
    }

    /* Setter dan Getter Supports Cuda */
    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public String getSupportsCuda() {
        return supportsCuda;
    }

    public void showMemoryData() {
        System.out.println("__/Product Data___________________");
        System.out.println("Price : " + this.getPrice());
        System.out.println("Id Product : " + this.getIdProduct());
        System.out.println("__________________________________");
        System.out.println(" ");
        System.out.println("__/Hardware Data___________________");
        System.out.println("Brand : " + this.getBrand());
        System.out.println("Model : " + this.getModel());
        System.out.println("__________________________________");
        System.out.println(" ");
        System.out.println("__/Memory Data____________________");
        System.out.println("Frequency : " + this.getFrequency());
        System.out.println("Memory Size : " + this.getMemorySize());
        System.out.println("Supports Cuda : " + this.getSupportsCuda());
        System.out.println("__________________________________");
        System.out.println(" ");
    }

}